flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/SingleCheckpointBarrierHandler.java

```diff
@@ -158,14 +163,16 @@ public void processBarrierAnnouncement(
 	@Override
 	public void processCancellationBarrier(CancelCheckpointMarker cancelBarrier) throws IOException {
 		final long cancelledId = cancelBarrier.getCheckpointId();
-		if (currentCheckpointId > cancelledId || (currentCheckpointId == cancelledId && numBarriersReceived == 0)) {
-			return;
+		if (cancelledId >= currentCheckpointId && (cancelledId > currentCheckpointId || numBarriersReceived > 0)) {
```

Comment:
Please double-check this @AHeise  :)

Reply:
It's correct but it might be easier to do
```
cancelledId > currentCheckpointId || (currentCheckpointId == cancelledId > numBarriersReceived == 0)
```
So either it's a future checkpoint, or it's the current checkpoint and not yet canceled.

Reply:
I like it!